### PR TITLE
ADEN-9849 Add types to spec

### PR DESCRIPTION
**Pull request by Bielik20**

### CODE REVIEW

File: spec/ad-engine/services/pbjs.stub.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 import { SinonSandbox, SinonStub } from 'sinon';
 
-export type PbjsStub = { [key in keyof Pbjs]: SinonStub };
+export type PbjsStub = { [key in keyof Pbjs]: SinonStub & Pbjs[key] };
 
 export function createPbjsStub(sandbox: SinonSandbox): PbjsStub {
 	return {
-		bidderSettings: sandbox.stub(),
-		adUnits: sandbox.stub(),
+		bidderSettings: sandbox.stub() as any,
```
Comment (jacekbj) [inline spec/ad-engine/services/pbjs.stub.ts]: Curious: why is that `any` necessary?

File: spec/ad-engine/services/pbjs.stub.ts [hunk 2]
```diff
@@ -25,7 +26,7 @@ export function stubPbjs(
 	sandbox: SinonSandbox,
 	pbjsStub: PbjsStub = createPbjsStub(sandbox),
 ): { pbjsInitStub: SinonStub; pbjsStub: PbjsStub } {
-	const pbjsInitStub = sandbox.stub(pbjsFactory, 'init').returns(Promise.resolve(pbjsStub));
+	const pbjsInitStub = sandbox.stub(pbjsFactory, 'init').resolves(pbjsStub);
```
Comment (jacekbj) [inline spec/ad-engine/services/pbjs.stub.ts]: 💎

File: spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts [hunk 3]
```diff
@@ -16,12 +17,14 @@ function getSlotElement() {
...
-function getAdSlotObject() {
-	return {
-		getElement: () => getSlotElement(),
-		getSlotName: () => 'top_leaderboard',
-		isEnabled: () => true,
-	};
+function getAdSlotObject(sandbox: SinonSandbox): AdSlotStub {
+	const stub = createAdSlotStub(sandbox);
+
+	stub.getElement.callsFake(getSlotElement);
```
Comment (jacekbj) [inline spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts]: 🔷

File: spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts [hunk 4]
```diff
@@ -103,49 +106,43 @@ function getParams() {
...
-
-		slotTweaker.onReady.returns({ then: () => {} });
-		slotTweaker.makeResponsive.returns({ then: () => {} });
-		global.window.scrollBy.returns(null);
-		global.window.addEventListener.returns(null);
+		sandbox.stub(slotTweaker, 'onReady').resolves({} as any);
+		sandbox.stub(slotTweaker, 'makeResponsive').resolves({} as any);
+		sandbox.stub(window, 'scrollBy').returns(null);
+		sandbox.stub(window, 'addEventListener').returns(null);
+		(document.body as any).style = new CSSStyleDeclaration();
```
Comment (jacekbj) [inline spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts]: Would that work?
```suggestion
		(document.body as HTMLBodyElement).style = new CSSStyleDeclaration();
```

File: spec/ad-engine/services/pbjs.stub.ts (hunk 1 above)
Comment (Bielik20) [reply spec/ad-engine/services/pbjs.stub.ts]: Necessary for fields :/

File: spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts (hunk 4 above)
Comment (Bielik20) [reply spec/ad-products/templates/uap/themes/hivi/hivi.spec.ts]: Maybe I am not sure.

**Pull request merged**